tests/web/Dim.Web.Tests/DimBusinessLogicTests.cs

```diff
@@ -425,4 +428,303 @@ private WalletData GetWalletData()
+        async Task Act() => await _sut.GetTechnicalUserProcess(bpn, companyName, technicalUserName);
+
+        // Assert
+        var exception = await Assert.ThrowsAsync<NotFoundException>(Act);
+        exception.Message.Should().Be($"No process data found for technical user {technicalUserName}");
+        A.CallTo(() => _technicalUserRepository.GetTechnicalUserProcess(bpn, companyName, technicalUserName))
+            .MustHaveHappenedOnceExactly();
+    }
+
+    [Fact]
+    public async Task GetTechnicalUserProcess_WithNonExistingProcess_ThrowsNotFoundException()
+    {
+        // Arrange
+        const string Bpn = "BPNL00000001TEST";
+        const string CompanyName = "testCompany";
+        const string TechnicalUserName = "nonExistingTechUser";
+
+        A.CallTo(() => _technicalUserRepository.GetTechnicalUserProcess(Bpn, CompanyName, TechnicalUserName))
+            .Returns(Task.FromResult<ProcessData?>(null));
+
+        // Act
+        async Task Act() => await _sut.GetTechnicalUserProcess(Bpn, CompanyName, TechnicalUserName);
+
+        // Assert
+        var exception = await Assert.ThrowsAsync<NotFoundException>(Act);
+        exception.Message.Should().Be($"No process data found for technical user {TechnicalUserName}");
+        A.CallTo(() => _technicalUserRepository.GetTechnicalUserProcess(Bpn, CompanyName, TechnicalUserName))
+            .MustHaveHappenedOnceExactly();
+    }
+
+    [Theory]
+    [InlineData("BPNL00000001TEST", "testCompany1", "testUser1")]
+    [InlineData("BPNL00000002TEST", "testCompany2", "testUser2")]
+    [InlineData("BPNL00000003TEST", "testCompany3", "testUser3")]
+    public async Task GetTechnicalUserProcess_WithDifferentValidInputs_ReturnsCorrectProcessData(string bpn, string companyName, string technicalUserName)
+    {
+        // Arrange
+        var expectedProcessData = _fixture.Create<ProcessData>();
+
+        A.CallTo(() => _technicalUserRepository.GetTechnicalUserProcess(bpn, companyName, technicalUserName))
+            .Returns(expectedProcessData);
+
+        // Act
+        var result = await _sut.GetTechnicalUserProcess(bpn, companyName, technicalUserName);
+
+        // Assert
+        result.Should().BeEquivalentTo(expectedProcessData);
+        A.CallTo(() => _technicalUserRepository.GetTechnicalUserProcess(bpn, companyName, technicalUserName))
+            .MustHaveHappenedOnceExactly();
+    }
+
+    #endregion
+
+    #region RetriggerProcess
+
+    [Theory]
+    [InlineData(ProcessTypeId.SETUP_DIM, ProcessStepTypeId.RETRIGGER_CREATE_WALLET)]
+    [InlineData(ProcessTypeId.SETUP_DIM, ProcessStepTypeId.RETRIGGER_CHECK_OPERATION)]
+    [InlineData(ProcessTypeId.TECHNICAL_USER, ProcessStepTypeId.RETRIGGER_CREATE_TECHNICAL_USER)]
+    [InlineData(ProcessTypeId.TECHNICAL_USER, ProcessStepTypeId.RETRIGGER_DELETE_TECHNICAL_USER)]
+    public async Task RetriggerProcess_WithValidData_ExecutesSuccessfully(ProcessTypeId processTypeId, ProcessStepTypeId processStepTypeId)
```

Comment:
Could you please add some checks if the new ProcessSteps are created correctly? And that the existing retrigger step is set to done.

Reply:
Thanks for the review. Added the checks.